encryption/encryption.cc

```diff
@@ -265,6 +266,16 @@ Status KeyManagedEncryptedEnv::NewSequentialFile(
     case EncryptionMethod::kAES192_CTR:
     case EncryptionMethod::kAES256_CTR:
       s = encrypted_env_->NewSequentialFile(fname, result, options);
+      if (!s.ok()) {
+        return s;
+      }
+      // Hack: sometimes the current file may be read as an encrypted
+      // file by mistakes.
+      if (ShouldSkipEncryption(fname) && !isValidCurrentFile(result)) {
```

Comment:
btw let's have this treatment for the CURRENT file only. Replace `ShouldSkipEncryption(fname)` with a new `IsCurerntFile(fname)` here.